### PR TITLE
Open generated docs

### LINKED ISSUE #379: Serve generated docs
When doing `spago docs` one has to start a webserver in order to get the nice searchbar, and that adds friction.

Instead we should serve the docs with [wai-app-static](https://www.stackage.org/package/wai-app-static)

### PR DESCRIPTION
**Pull request by DStranger**
### Description of the change

Fixes #379.
Print a link to the `index.html` of the generated docs.
Add an `--open` flag to `spago docs` which opens them in browser upon generation.

Not sure how this can be tested though 🙂

### Checklist:

- [x] Added the change to the "Unreleased" section of the changelog
- [x] Added some example of the new feature to the `README`
- [ ] Added a test for the contribution (if applicable)

### CODE REVIEW

File: app/Spago.hs [hunk 1]
```diff
@@ -159,6 +160,7 @@ parser = do
     jsonFlag    = bool JsonOutputNo JsonOutputYes <$> CLI.switch "json" 'j' "Produce JSON output"
     dryRun      = bool DryRun NoDryRun <$> CLI.switch "no-dry-run" 'f' "Actually perform side-effects (the default is to describe what would be done)"
     usePsa      = bool UsePsa NoPsa <$> CLI.switch "no-psa" 'P' "Don't build with `psa`, but use `purs`"
+    openDocs    = bool NoOpenDocs DoOpenDocs <$> CLI.switch "open" 'o' "Open generated documentation in browswer (for HTML format only)"
```
[the remark this answers is not in the document]
Comment (klntsky) [reply app/Spago.hs]: > it could be nice to just open the browser by default

Please noooo

[the remark this answers is not in the document]
Comment (klntsky) [reply app/Spago.hs]: It introduces a lot of assumptions (i.e. that a browser is installed, we have xdg thingy configured properly, and the user is actually generating the docs for immediate use).

[the remark this answers is not in the document]
Comment (DStranger) [reply app/Spago.hs]: Yes, I also think it should be opt-in

File: src/Spago/Build.hs [hunk 2]
```diff
@@ -213,36 +216,54 @@ bundleModule maybeModuleName maybeTargetPath noBuild buildOpts = do
...
+  when isHTMLFormat $ do
+    link <- linkToIndexHtml
+    let linkText = "Link: " <> link
+    echo linkText
+
+    when (open == DoOpenDocs) $ do
+      echo "Opening in browser..."
+      () <$ openLink link
+
+    when (noSearch == AddSearch) $ do
```
[the remark this answers is not in the document]
Comment (DStranger) [reply src/Spago/Build.hs]: Will do!

File: app/Spago.hs (hunk 1 above)
[the remark this answers is not in the document]
Comment (f-f) [reply app/Spago.hs]: > i.e. that a browser is installed, we have xdg thingy configured properly, and the user is actually generating the docs for immediate use

And these conditions are met in like 90% of the cases?

To properly answer the question of "should we make this a default" we'd need data of course, but at least in my team the vast majority of times we generate docs is for immediate use and on a "properly configured desktop/laptop"

The only case I can think of when you don't want to generate docs for immediate use is in CI

[the remark this answers is not in the document]
Comment (f-f) [reply app/Spago.hs]: But alright, let's keep it opt-in 🙂

File: CHANGELOG.md [hunk 3]
```diff
@@ -43,6 +43,10 @@ Other Improvements:
 - Deps: upgrade to dhall-1.26.0, v10 of the standard (#411, #358)
 
 
+New features:
```
[the remark this answers is not in the document]
Comment (DStranger) [reply CHANGELOG.md]: Oh, true, that must've been github's 'merge with master' button

**Comment by DStranger** [comment]
> @DStranger thank you for doing this! 👏

I'm glad I can help :)

**Pull request merged**